magda/pipeline/graph.py

```diff
@@ -97,19 +97,33 @@ async def run(
                     request=MagdaLogger.Parts.Request(str(request)),
                 )
                 logger.event('RUN')
-                module_result = await self._run_method_helper(
-                    module, data, request, is_regular_runtime, logger,
-                )
-                results.append(
-                    Module.Result(
-                        result=module_result,
-                        interface=module.interface,
-                        name=module.name,
-                        src_class=type(module),
-                        expose=self._exposed_result(module, is_regular_runtime),
+                try:
+                    module_result = await self._run_method_helper(
+                        module, predecessors_data, request, is_regular_runtime, logger,
                     )
-                )
-        return results
+                    results.append(
+                        Module.Result(
+                            result=module_result,
+                            error=None,
+                            interface=module.interface,
+                            name=module.name,
+                            src_class=type(module),
+                            expose=self._exposed_result(module, is_regular_runtime),
+                        )
+                    )
+                except Exception as e:
+                    logger.info(f"Module {module.name} has failed. Stopping the pipeline")
```

Comment:
`logger` has all information about pipeline and module (lines 92-96) so you don't have to write them again. However, logging the error might be useful. So, I'd recommend to wait for #20 and change to:
```suggestion
                    logger.error(e)
```

Reply:
Added a TODO